source/includes/_oauth2.md

```diff
@@ -42,6 +42,208 @@ In this page they can grant access to a data partner by selecting the data partn
+  "expires_in":7200,
+  "scope":"write:assets"
+}
+```
+
+```shell
+curl \
+  -F "access_token=$ACCESS_TOKEN" \
+  -X POST https://api.gresb.com/oauth/token/info?
+```
+
+> Response:
+
+```json
+{
+  "resource_owner_id":1234,
+  "scope":["public","write:assets"],
+  "expires_in":7200,
+  "application": {
+    "uid":"d26b1521a034a2e4cfc6372b0db51e6790421e1fd00fd97d5a0d923fe67b685b"
+  },
+  "created_at":1548944742
+}
+```
+
+You can now request an access token by issuing a POST request to
+`/oauth/token`.  You must include `grant_type=authorization_code`, your
+`client_id`, `client_secret`, and the authorization `code` as parameters to
+your request.  In return you will receive an `access_token` for you
+application.
+
+`POST /oauth/token?client_id=$CLIENT_ID&client_secret=$CLIENT_SECRET&code=$AUTHORIZATION_CODE&redirect_uri=urn:ietf:wg:oauth:2.0:oob`
+
+
+#### Step 3. Use Access Token
+
+The access token can now be used to make requests to the GRESB API.  The token must be included either as a `Authorization: Bearer` HTTP header, or as a request parameter named `access_token`.
+
+As an HTTP Header:
+
+`curl https://api.gresb.com/api/responses -H 'Authorization: Bearer $ACCESS_TOKEN'`
+
+As a request parameter:
+
+`curl https://api.gresb.com/api/responses?access_token=$ACCESS_TOKEN`
+
+### OAuth Errors
+
+**Common**
+
+* **invalid_request**: 'The request is missing a required parameter, includes an unsupported parameter - value, or is otherwise malformed.'
+* **invalid_redirect_uri**: 'The redirect uri included is not valid.'
+* **unauthorized_client**: 'The client is not authorized to perform this request using this method.'
+* **access_denied**: 'The resource owner or authorization server denied the request.'
+* **invalid_scope**: 'The requested scope is invalid, unknown, or malformed.'
+* **server_error**: 'The authorization server encountered an unexpected condition which prevented it from fulfilling the request.'
+* **temporarily_unavailable**: 'The authorization server is currently unable to handle the request due to a temporary overloading or maintenance of the server.'
+
+**Access Grant Errors**
+
+* **unsupported_response_type**: 'The authorization server does not support this response type.'
+
+**Access Token Errors**
+
+* **invalid_client**: 'Client authentication failed due to unknown client, no client authentication included, or unsupported authentication method.'
+* **invalid_grant**: 'The provided authorization grant is invalid, expired, revoked, does not match the redirection URI used in the authorization request, or was issued to another client.'
+* **unsupported_grant_type**: 'The authorization grant type is not supported by the authorization server.'
+
+**Password Access Token Errors**
+
+* **invalid_resource_owner**: 'The provided resource owner credentials are not valid, or resource owner cannot be found'
+
+**Invalid Tokens**
+
+* **revoked**: "The access token was revoked"
+* **expired**: "The access token expired"
```

Comment:
I don't think this can happen. Since GAMS was introduced we no longer communicate whether a token is expired, after a certain time it's just no longer found

Reply:
makes sense - I'll remove it